redis.go

```diff
@@ -51,39 +55,39 @@ func (ds *Datastore) Put(key datastore.Key, value []byte) error {
 	return nil
 }
 
-func (ds *Datastore) Sync(prefix datastore.Key) error {
+func (ds *Datastore) Sync(ctx context.Context, prefix datastore.Key) error {
 	return nil
 }
 
-func (ds *Datastore) Get(key datastore.Key) (value []byte, err error) {
+func (ds *Datastore) Get(ctx context.Context, key datastore.Key) (value []byte, err error) {
 	ds.mu.Lock()
 	defer ds.mu.Unlock()
 	return ds.client.Cmd("GET", key.String()).Bytes()
 }
 
-func (ds *Datastore) GetSize(key datastore.Key) (size int, err error) {
+func (ds *Datastore) GetSize(ctx context.Context, key datastore.Key) (size int, err error) {
 	ds.mu.Lock()
 	defer ds.mu.Unlock()
 	return ds.client.Cmd("STRLEN", key.String()).Int()
 }
 
-func (ds *Datastore) Has(key datastore.Key) (exists bool, err error) {
+func (ds *Datastore) Has(ctx context.Context, key datastore.Key) (exists bool, err error) {
 	ds.mu.Lock()
 	defer ds.mu.Unlock()
 	return ds.client.Cmd("EXISTS", key.String()).Bool()
 }
 
-func (ds *Datastore) Delete(key datastore.Key) (err error) {
+func (ds *Datastore) Delete(ctx context.Context, key datastore.Key) (err error) {
 	ds.mu.Lock()
 	defer ds.mu.Unlock()
 	return ds.client.Cmd("DEL", key.String()).Err
 }
 
-func (ds *Datastore) Query(q query.Query) (query.Results, error) {
+func (ds *Datastore) Query(ctx context.Context, q query.Query) (query.Results, error) {
 	return nil, errors.New("TODO implement query for redis datastore?")
 }
 
-func (ds *Datastore) Batch() (datastore.Batch, error) {
+func (ds *Datastore) Batch(ctx context.Context) (datastore.Batch, error) {
 	return nil, datastore.ErrBatchUnsupported
 }
```

Comment:
This is sort of silly 🙄, but we don't have to deal with this now. It seems like this error is standard in parts of the codebase